docs/developer-docs/web-apps/verifiable-credentials/issuer.mdx

```diff
@@ -0,0 +1,184 @@
+
+An example implementation in Rust can be found below:
+
+```rust
+pub fn get_vc_consent_message_en(
+    credential_spec: &CredentialSpec,
+) -> Result<Icrc21ConsentInfo, Icrc21Error> {
+    match verify_spec_and_get_group_name(credential_spec) {
+        Err(err) => Err(Icrc21Error::ConsentMessageUnavailable(Icrc21ErrorInfo {
+            description: err,
+        })),
+        Ok((group_name, _owner)) => Ok(Icrc21ConsentInfo {
+            consent_message: format!("# \"{}\"", group_name),
+            language: "en".to_string(),
+        }),
+    }
+}
+```
+
+If the response is an approval, the identity provider returns the consent message to the issuer.
+
+### Step 2: Implement a derivation origin (`derivation_origin`)
+
+Next, the issuer API endpoint `derivation_origin` enables support of the [alternative derivation origins](https://internetcomputer.org/docs/current/references/ii-spec#alternative-frontend-origins) feature.
+
+A derivation origin is an alternative URL that can be obtained by the identity provider for the user's principal. If the issuer does not implement alternative derivation origins, the identity provider will return the default value of the canister's URL (`https://<issuer-canister-id>.icp0.io`).
+
+This endpoint verifies that a user's principal, if issued by an alternative origin, is valid and can be issued a verifiable credential.
+
+An example implementation in Rust can be found below:
+
+```rust
+async fn derivation_origin(
+    req: DerivationOriginRequest,
+) -> Result<DerivationOriginData, DerivationOriginError> {
+    get_derivation_origin(&req.frontend_hostname)
+}
+
+fn get_derivation_origin(_hostname: &str) -> Result<DerivationOriginData, DerivationOriginError> {
+    CONFIG.with_borrow(|config| {
+        let config = config.get();
+        println!("*** derivation origin: {}", config.derivation_origin);
+        Ok(DerivationOriginData {
+            origin: config.derivation_origin.clone(),
+        })
+    })
+}
+```
+
+:::info
+The returned derivation origin is subject to verification via `.well-known/ii-alternative-origins`, as described in the [feature description](https://internetcomputer.org/docs/current/references/ii-spec#alternative-frontend-origins).
+:::
+
+### Step 3: Prepare the credentials (`prepare_credential`)
+
+Before a credential can be issued, the request for the credential must be checked for validity using the `prepare_credential` API endpoint. Once it has been validated, then the actual VC can be prepared. The request is validated through the `signed_id_alias.credential_jws` token. If this check is successful, the credential data is prepared and returned in `PreparedCredentialData.prepared_context`.
+
+An example implementation in Rust can be found below:
+
+```rust
```

Comment:
This is incomplete.

I would point to the [VC playground code](https://github.com/dfinity/vc-playground/blob/b2f8cd5f5397b7b355b6503df5af3789834b79e9/issuer/src/main.rs#L557).

The code is too long to added here.

Reply:
Replaced code with link to the playground code